pkg/scaling/scale_handler.go

```diff
@@ -293,23 +293,20 @@ func (h *scaleHandler) getScalersCacheForScaledObject(ctx context.Context, scale
 // performGetScalersCache returns cache for input scalableObject, it is common code used by GetScalersCache() and getScalersCacheForScaledObject() methods
 func (h *scaleHandler) performGetScalersCache(ctx context.Context, key string, scalableObject interface{}, scalableObjectGeneration *int64, scalableObjectKind, scalableObjectNamespace, scalableObjectName string) (*cache.ScalersCache, error) {
 	h.scalerCachesLock.RLock()
-	regenerateCache := false
+
 	if cache, ok := h.scalerCaches[key]; ok {
 		// generation was specified -> let's include it in the check as well
 		if scalableObjectGeneration != nil {
 			if cache.ScalableObjectGeneration == *scalableObjectGeneration {
 				h.scalerCachesLock.RUnlock()
 				return cache, nil
 			}
-			// object was found in cache, but the generation is not correct,
-			// we'll need to close scalers in the cache and
-			// proceed further to recreate the cache
-			regenerateCache = false
```

Comment:
I think the logically was wrong originally. I think it should have been `regenerateCache = true` since current cache entry is about to be regenerated.

I do not think it was necessary to decide there if a regeneration was necessary. It can be delayed until we enter the next mutex protected block.